### PR TITLE
Set component state to healthy after registe session done

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #14077 
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/datacoord/server.go [hunk 1]
```diff
@@ -295,12 +296,23 @@ func (s *Server) Start() error {
...
-	logutil.Logger(s.ctx).Debug("startup success")
 
 	return nil
 }
 
+// UpdateStateCode updates dataCoord's state code
+// TODO:: use internalpb.StateCode instead of ServerState
+func (s *Server) UpdateStateCode(code internalpb.StateCode) {
+	switch code {
+	case internalpb.StateCode_Initializing:
```
Comment (cydrain) [inline internal/datacoord/server.go:307]: const (
	StateCode_Initializing StateCode = 0
	StateCode_Healthy      StateCode = 1
	StateCode_Abnormal     StateCode = 2
)
I think above variables are redundant, why not use following variables directly?

const (
	// ServerStateStopped state stands for just created or stopped `Server` instance
	ServerStateStopped ServerState = 0
	// ServerStateInitializing state stands initializing `Server` instance
	ServerStateInitializing ServerState = 1
	// ServerStateHealthy state stands for healthy `Server` instance
	ServerStateHealthy ServerState = 2
)

**Comment by xige-16** [comment]
During the startup process of a certain component, it is still unavoidable that grpc returns an error, and each component needs to carefully handle the grpc error

**Pull request closed without merging**